src/main/java/io/nats/client/BaseConsumerContext.java

```diff
@@ -18,6 +18,7 @@
 
 /**
  * The Consumer Context provides a convenient interface around a defined JetStream Consumer
+ * <p> Note: ConsumerContext requires a <b>pull consumer</b>. 
```

Comment:
So this may eventually not be true. For instance I know Alberto and javascript are working on supporting push consumers. And also, it takes effort to make a push consumer because you have to specifically add the delivery subject. I'm not sure if this is necessary, so maybe just let me know what led you to add this? I guess we can go back if we do support push

Reply:
It is not obvious that that push consumers don't work and can result in late errors, late during application deployment. This is driven by a lot of mix-ups between push and pull consumer I see from user in slack.
Reading just the documentation I initially assumed that ConsumerContext would support push and pull and was quite a bit surprised.